### PR TITLE
Render Basic calendar

### PR DESCRIPTION
**Pull request by YesImKenny**
Need to still work on positioning of the two calendars
Screen Shot 2019-05-23 at 9 58 56 AM

### CODE REVIEW

File: client/src/components/Calendar.jsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+  firstDayOfMonth(leftOrRight) {
+    let firstDay = moment(leftOrRight).startOf('month').format('d');
+    return firstDay;
+  }
+
+  render() {
+    const eachDayOfWeek = moment.weekdaysShort().map(day => (
+      <DayOfMonth key={day}>
+        {day.slice(0, -1)}
```
Comment (EtiWitty) [inline client/src/components/Calendar.jsx]: it's hard for me to grasp what you are doing here.  Do you mind adding some comment or changing the variable names?

Given "slice" is usually called on an Array/String, I'm having hard time guessing what is stored in a variable called `day` as an `Array/String`.

File: client/src/components/Calendar.jsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        <CalendarDay style={{ border: 'none' }}></CalendarDay>
+      );
+    }
+
+    let rightFiller = [];
+    for (let i = 0; i < this.firstDayOfMonth(this.state.rightMonth); i += 1) {
+      rightFiller.push(
+        <CalendarDay style={{ border: 'none' }}></CalendarDay>
+      );
+    }
```
Comment (EtiWitty) [inline client/src/components/Calendar.jsx]: I guess `let rightFiller = this.firstDayOfMonth(this.state.rightMonth).map(...)` could make it easier.  Same for `leftFiller`.

File: client/src/components/Calendar.jsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        daysPerEachWeekRight = [];
+        daysPerEachWeekRight.push(row);
+      }
+      if (i === totalRightCalendar.length - 1) {
+        rowsOfDaysRight.push(daysPerEachWeekRight);
+      }
+    });
+
+    let daysinLeftmonth = rowsOfDaysLeft.map(d => <tr>{d}</tr>);
+    let daysinRightmonth = rowsOfDaysRight.map(d => <tr>{d}</tr>);
```
Comment (EtiWitty) [inline client/src/components/Calendar.jsx]: This whole `render` function is getting a little bit long and hard to follow the logic and tasks done in here.  It might not be a bad idea to have one or more functions to put this logic there. Maybe a simple `preprocess()` function.

Also, another thing to consider is `render` function in React can be called several times.  So, while performance might not be a concern to us right now, it would be a healthy practice to keep render as lightweight as possible, by shifting some one-time-only calculations to the constructor or other methods in the lifecycle that are less frequent than `render` (e.g. `componentDidMount`.

File: client/src/components/Calendar.jsx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  font-style: bold;
+  padding-left: 225px;
+  width: 22%;
+  margin: 5em;
+  border-spacing: 0;
+  border-collapse: collapse;
+`;
+
+const Weekday = styled.section`
+  padding: 20px 2;
```
Comment (EtiWitty) [inline client/src/components/Calendar.jsx]: I guess you mean `2px` here, right?

File: client/src/components/Calendar.jsx (hunk 3 above)
Comment (YesImKenny) [reply client/src/components/Calendar.jsx]: Thank you for the thorough review. I will implement these changes!

File: client/src/components/Calendar.jsx (hunk 1 above)
Comment (YesImKenny) [reply client/src/components/Calendar.jsx]: Thank you for pointing this out.
